### PR TITLE
Recipe search

### PR DESCRIPTION
**Pull request by Lexseal**
### Features Added/Issues Addressed
* added dynamic refresh based on the filter on the search page
* allowed the enter key to function as the search button

### Brief Description of Solution
Created a global variable to store the fetched recipes and display a subset of the global recipes when the checkboxes are toggled.
Additionally, added the search function as an event listener for the enter key keyup event.

### Checklist
- [ ] Added new tests (You know the code best to make new tests)
- [x] Resolved merge conflicts (Prefer merge over rebase)
- [x] Status check passed (Linted and tested)
- [ ] Reviewed (At least one reviewer's approval)

### Caveat/Special Instructions for Team
None

### CODE REVIEW

File: source/assets/scripts/backend.js [hunk 1]
```diff
@@ -66,7 +66,8 @@ function recalculate_intolerances(recipe){
       }
     }
   }
-  // if (recipe.intolerances.includes('Soy-free')) tag['Soy-free']=true; 
+  // if (recipe.intolerances.includes('Soy-free')) tag['Soy-free']=true;
+  if (tag['Vegan']) tag['Vegetarian'] = true;
```
Comment (Lexseal) [inline source/assets/scripts/backend.js:70]: saw a bunch of instances where a vegan thing isn't vegetarian so decided to fix it

File: source/assets/scripts/recipe-searchPage.js [hunk 2]
```diff
@@ -49,26 +51,38 @@ function bindSearchButton() {
   //once the button got clicked, request the data from api and then output the result
   let button = document.querySelector('#search-button');
   button.addEventListener('click', hitSearch);
+  // alternatively, enter key also trigger a search
+  const input_field = document.getElementById('search-field');
+  input_field.addEventListener('keyup', e => {
+    if (e.key === 'Enter') hitSearch();  // enter key pressed
```
Comment (Lexseal) [inline source/assets/scripts/recipe-searchPage.js:57]: this fixes the enter key not searching

File: source/assets/scripts/recipe-searchPage.js [hunk 3]
```diff
@@ -49,26 +51,38 @@ function bindSearchButton() {
...
+  });
+}
+
+/**
+ * displays the current recipes with filters applied
+ */
+function displayCards() {
```
Comment (Lexseal) [inline source/assets/scripts/recipe-searchPage.js:64]: factored this function outside so we can call when checkboxes are toggled

File: source/assets/scripts/recipe-searchPage.js [hunk 4]
```diff
@@ -4,12 +4,14 @@ import * as backend from './backend.js';
 
 window.addEventListener('DOMContentLoaded', init);
 
+let current_recipes = [];
```
Comment (Lexseal) [inline source/assets/scripts/recipe-searchPage.js:7]: this is all the recipes we have fetched from the backend

**Review by yiw008 — CHANGES_REQUESTED** [review]
/

File: source/assets/scripts/recipe-detail.js [hunk 5]
```diff
@@ -91,9 +91,9 @@ function bindFoodieButton() {
...
 function goBack(){
   const btn = document.getElementById('white-arrow-p');
-  let index = document.referrer.lastIndexOf('/');
-  let str = document.referrer.substring(index + 1);
-  console.log(str);
+  // let index = document.referrer.lastIndexOf('/');
+  // let str = document.referrer.substring(index + 1);
+  // console.log(str);
 
   btn.addEventListener('click', () => window.history.back());
```
Comment (yiw008) [inline source/assets/scripts/recipe-detail.js:98]: For a saved recipe, after I entered the recipe from dashboard, clicked the saved button (cancel favorite), and go back to dashboard, the recipe is still in the saved recipe list. That's why I need to check whether str is index.html.

**Review by yiw008 — APPROVED** [review]
Changed the goBack function in recipe-detail.js

**Comment by yiw008** [comment]
For a saved recipe, after I entered the recipe from dashboard, clicked the saved button (cancel favorite), and go back to dashboard, the recipe is still in the saved recipe list. To solve this problem, I need to check the str.

**Pull request merged**